spec/factories/vm_or_template.rb

```diff
@@ -14,10 +14,59 @@
     raw_power_state "never"
   end
 
-  factory(:vm,             :class => "Vm",            :parent => :vm_or_template)
-  factory(:vm_cloud,       :class => "VmCloud",       :parent => :vm)       { cloud true }
-  factory(:vm_infra,       :class => "VmInfra",       :parent => :vm)
+  factory(:vm, :class => "Vm", :parent => :vm_or_template)
+
+  factory(:vm_orphaned, :class => "Vm", :parent => :vm_or_template) do
+    ext_management_system nil
+    storage_id            123
+  end
+
+  factory(:template_orphaned, :class => "TemplateInfra", :parent => :vm_or_template) do
+    ext_management_system nil
+    storage_id            123
+  end
+
+  factory(:vm_cloud_with_az, :class => "VmCloud", :parent => :vm) do
+    ext_management_system { FactoryGirl.create(:ems_google) }
+    storage_id            123
+    availability_zone     { FactoryGirl.create(:availability_zone_google) }
+  end
+
+  factory(:vm_cloud_without_az, :class => "VmCloud", :parent => :vm) do
```

Comment:
Defining `vm_cloud_with_az` works.
But don't define a factory by what it doesn't have.

Just call this `vm_cloud` - think that was in place before.